app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/environment/EnvironmentProductFragment.java

```diff
@@ -10,41 +12,102 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
 
+import com.squareup.picasso.Picasso;
+
+import java.io.File;
+
+import io.reactivex.disposables.CompositeDisposable;
 import openfoodfacts.github.scrachx.openfood.R;
 import openfoodfacts.github.scrachx.openfood.databinding.FragmentEnvironmentProductBinding;
+import openfoodfacts.github.scrachx.openfood.features.FullScreenActivityOpener;
+import openfoodfacts.github.scrachx.openfood.features.ImagesManageActivity;
+import openfoodfacts.github.scrachx.openfood.features.product.edit.ProductEditActivity;
+import openfoodfacts.github.scrachx.openfood.features.product.view.ProductViewActivity;
 import openfoodfacts.github.scrachx.openfood.features.shared.BaseFragment;
+import openfoodfacts.github.scrachx.openfood.images.ProductImage;
 import openfoodfacts.github.scrachx.openfood.models.Nutriments;
 import openfoodfacts.github.scrachx.openfood.models.Product;
 import openfoodfacts.github.scrachx.openfood.models.ProductState;
+import openfoodfacts.github.scrachx.openfood.network.OpenFoodAPIClient;
 import openfoodfacts.github.scrachx.openfood.utils.FragmentUtils;
+import openfoodfacts.github.scrachx.openfood.utils.LocaleHelper;
+import openfoodfacts.github.scrachx.openfood.utils.PhotoReceiverHandler;
+import openfoodfacts.github.scrachx.openfood.utils.Utils;
 
+import static openfoodfacts.github.scrachx.openfood.models.ProductImageField.PACKAGING;
 import static openfoodfacts.github.scrachx.openfood.utils.Utils.bold;
+import static org.apache.commons.lang.StringUtils.isNotBlank;
 
 public class EnvironmentProductFragment extends BaseFragment {
+    private static final int EDIT_PRODUCT_AFTER_LOGIN_REQUEST_CODE = 1;
+    private CompositeDisposable disp;
+    private PhotoReceiverHandler photoReceiverHandler;
     private FragmentEnvironmentProductBinding binding;
     private ProductState activityProductState;
+    private String mUrlImage;
+    private String barcode;
+    private OpenFoodAPIClient api;
+    /**
+     * boolean to determine if image should be loaded or not
+     **/
+    private boolean isLowBatteryMode = false;
+    private Product product;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        disp = new CompositeDisposable();
```

Comment:
I would inizialize this in the declaration and make it final.

Reply:
Cool. But wouldn't it be better to use Uber's [AutoDispose](https://uber.github.io/AutoDispose) and get rid of all the CompositeDisposables throughout the app?